src/v/cluster/data_migration_backend.cc

```diff
@@ -564,11 +585,14 @@ ss::future<errc> backend::do_topic_work(
     }
 }
 
-void backend::abort_all_topic_work() {
+ss::future<> backend::abort_all_topic_work() {
     for (auto& [nt, tsws] : _active_topic_work_states) {
         tsws->rcn.request_abort();
     }
-    _active_topic_work_states.clear();
+    while (!_active_topic_work_states.empty()) {
+        co_await _active_topic_work_states.begin()
+          ->second->promise.get_future();
+    }
```

Comment:
this seems very confusing, can we just wait for migration to finish in the first loop ?